app/src/main/java/io/github/androidacademyglobal/profile/ProfileScreen.kt

```diff
@@ -50,37 +55,48 @@ fun ProfileScreen(userData: ProfileScreenState) {
             .nestedScroll(scrollBehavior.nestedScrollConnection)
     ) {
         BoxWithConstraints(modifier = Modifier.weight(AppConstants.PROFILE_BOX_WEIGHT)) {
-            Surface {
-                Column(
-                    modifier = Modifier
-                        .fillMaxSize()
-                        .verticalScroll(scrollState),
-                ) {
-                    ProfileHeader(
-                        scrollState,
-                        userData,
-                        this@BoxWithConstraints.maxHeight
-                    )
-                    UserInfoFields(userData, this@BoxWithConstraints.maxHeight)
-                }
+            ProfileView(userData = userData) {
+                maxHeight
             }
         }
     }
 }
 
+@Composable
+internal fun ProfileView(
+    userData: ProfileScreenState,
+    containerHeightProvider: () -> Dp
```

Comment:
Why lambda is better that a simple `Dp` value?

Reply:
Weird, I thought Compose detects automatically - what to recompose (walking doing the tree of composable functions) but I haven't checked it, thanks for pointing out!       
@grishaster80 FYI ^, it's your code here being changed - it might be helpful when developing production code ;)

Reply:
It detects automatically, but if the value changes the whole function result is invalidated instead of one part